libafl/src/observers/stdio.rs

```diff
@@ -168,91 +169,58 @@ use crate::{observers::Observer, Error};
 /// ```
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub struct StdOutObserver {
+pub struct StreamObserver<T> {
     /// The name of the observer.
     pub name: Cow<'static, str>,
-    /// The stdout of the target during its last execution.
-    pub stdout: Option<Vec<u8>>,
+    /// The captured stdout/stderr data during last execution.
+    pub data: Option<Vec<u8>>,
+    /// Phantom data to hold the stream type
+    phantom: PhantomData<T>,
 }
 
-/// An observer that captures stdout of a target.
-impl StdOutObserver {
-    /// Create a new [`StdOutObserver`] with the given name.
+/// Marker traits to distinguish between stdout and stderr
+#[derive(Debug, Clone)]
+pub struct StdOutMarker;
+/// Marker traits to distinguish between stdout and stderr
```

Comment:
Either change the doc or reference `StdOutMarker`. Just reading the doc for this dude withotu context would feel weird